### ISSUE TITLE
binding-positions: Report bindings inside HTML comments

### ISSUE THREAD

**Issue by coreyfarrell** [43081j/eslint-plugin-lit#42]
Putting a binding inside an HTML comment is almost always an error:
```js
html`<!-- <div>${value}</div> -->`
```

It's clear in this case but sometimes people will use HTML comment's disable a large block inside a template.  ~~Doing so does not prevent the `${value}` values from being processed, in some cases the evaluation might even have side-effects~~.  I wrote [babel-plugin-template-html-minifier](https://github.com/cfware/babel-plugin-template-html-minifier) which captures this if `removeComments` is enabled because the binding location gets removed by html-minifier, but it would be great if this situation could be reported by eslint.

CC @LarsDenBakker

**Comment by 43081j** [comment]
are you sure this is valid in lit?

```ts
html`<!-- ${foo} -->`
```

results in mangled source for me (because lit inserts a placeholder comment, which closes the one we've written).

sounds to me like its in fact invalid/unsupported by lit but works for you because you strip it out beforehand?

**Comment by coreyfarrell** [comment]
Hmmm apparently it's not valid in `lighterhtml` either (sorry I missed this).  My babel plugin throws an exception if `removeComments` is enabled with `${bidning}` inside any comment, so it doesn't actually remove it, just reports it as an error.

So to answer your question, it appears it is not valid.  I think that supports the idea that it would be appropriate for `binding-positions` to report this?

**Comment by 43081j** [comment]
Yup of course, I guess i hadn't thought of it because it would produce an invalid template. But it could be good to capture it seeing as its possible some workflows strip them (making the error missable).

**Comment by LarsDenBakker** [comment]
Thanks guys!
